src/partition/repart.c

```diff
@@ -4281,11 +4281,15 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_pretty = r;
                         break;
 
-                case ARG_DEFINITIONS:
-                        r = parse_path_argument(optarg, false, &arg_definitions);
+                case ARG_DEFINITIONS: {
+                        _cleanup_free_ char *path = NULL;
+                        r = parse_path_argument(optarg, false, &path);
                         if (r < 0)
                                 return r;
+                        if (strv_extend(&arg_definitions, path) < 0)
```

Comment:
use strv_push() and TAKE_PTR afterwards to donate the allocated memory to the strv. i.e.

```c
if (strv_push(&arg_definitions, &path) < 0)
        return log_oom();

TAKE_PTR(path);
```

Reply:
What about 

```
if (strv_consume(&arg_definitions), TAKE_PTR(path)) < 0)
        return log_oom();
```

?

Reply:
You are right, of course!